test/e2e/l7networkpolicy_test.go

```diff
@@ -0,0 +1,233 @@
+
+	// Create a client Pod which will be selected by test L7 NetworkPolices.
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(clientPodLabels).Create(data))
+	if _, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", clientPodName, err)
+	}
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, clientPodName, data.testNamespace))
+
+	serverPodName := "test-l7-http-server"
+	serverPodLabels := map[string]string{"test-l7-http-e2e": "server"}
+	cmd = []string{"bash", "-c", "/agnhost netexec --http-port=8080"}
+	require.NoError(t, NewPodBuilder(serverPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(serverPodLabels).Create(data))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, serverPodName, data.testNamespace)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", serverPodName, err)
+	}
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, serverPodName, data.testNamespace))
+	var serverIPs []*net.IP
+	if podIPs.ipv4 != nil {
+		serverIPs = append(serverIPs, podIPs.ipv4)
+	}
+	if podIPs.ipv6 != nil {
+		serverIPs = append(serverIPs, podIPs.ipv6)
+	}
+
+	l7ProtocolPermitsPathHostname := []crdv1alpha1.L7Protocol{
+		{
+			HTTP: &crdv1alpha1.HTTPProtocol{
+				Method: "GET",
+				Path:   "/hostname",
+			},
+		},
+	}
+	l7ProtocolPermitsAnyPath := []crdv1alpha1.L7Protocol{
+		{
+			HTTP: &crdv1alpha1.HTTPProtocol{
+				Method: "GET",
+			},
+		},
+	}
+
+	policyPermitPathHTTP := "test-l7-http-permit-path-hostname"
+	policyPermitAnyPath := "test-l7-http-permit-any-path"
+
+	probeFn := func() {
+		for _, ip := range serverIPs {
+			baseURL := net.JoinHostPort(ip.String(), "8080")
+
+			// Path 'hostname' is allowed.
+			hostname, err := probeHostnameFromPod(data, clientPodName, agnhostContainerName, baseURL)
+			assert.NoError(t, err)
+			assert.Equal(t, serverPodName, hostname)
+
+			// Path 'clientip' is not allowed.
+			_, err = probeClientIPFromPod(data, clientPodName, agnhostContainerName, baseURL)
+			assert.NotNil(t, err)
+
+			// Non-HTTP connections should be rejected by Suricata.
+			if ip.To4() != nil {
+				cmd = []string{"bash", "-c", fmt.Sprintf("dig @%s google.com a +tcp -p 8080", ip)}
+				stdout, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, agnhostContainerName, cmd)
+				// For the client Pod which is selected by the L7 NetworkPolicy, the expected output returned
+				// from Suricata should contain "connection reset".
+				assert.NoError(t, err)
+				assert.Contains(t, stdout, fmt.Sprintf("communications error to %s#8080: connection reset", ip))
+			}
+		}
+
+		// Delete the ANP that allows HTTP path 'hostname'.
+		data.crdClient.CrdV1alpha1().NetworkPolicies(data.testNamespace).Delete(context.TODO(), policyPermitPathHTTP, metav1.DeleteOptions{})
```

Comment:
It looks weird to include deleting operations in `probeFn`. 
I think it should be something like:
```
createL7AntreaNetworkPolicy()
probeFn(true, true, false)
updateL7AntreaNetworkPolicy()
probeFn(true, false, false)
deleteL7AntreaNetworkPolicy()
probeFn(true, true, true)
```